src/pages/patch/patchTabs/tasks/TasksTable.tsx

```diff
@@ -62,6 +64,65 @@ const getSortDirFromOrder = (order: "ascend" | "descend") =>
 
 const rowKey = ({ id }: { id: string }): string => id;
 
+const mapTaskStatusToBadgeVariant = {
+  [TaskStatus.Inactive]: Variant.LightGray,
+  [TaskStatus.Unstarted]: Variant.LightGray,
+  [TaskStatus.Undispatched]: Variant.LightGray,
+  [TaskStatus.Started]: Variant.Yellow,
+  [TaskStatus.Dispatched]: Variant.LightGray,
+  [TaskStatus.Succeeded]: Variant.Green,
+  [TaskStatus.Failed]: Variant.Red,
+  [TaskStatus.StatusBlocked]: Variant.DarkGray,
+  [TaskStatus.StatusPending]: Variant.Yellow
+};
+
+const failureColors = {
+  text: "#800080",
+  border: "#CC99CC",
+  fill: "#E6CCE6"
+};
+
+// the status colors that are not supported by the leafygreen Badge variants
+const mapUnsupportedBadgeColors = {
+  [TaskStatus.SystemFailed]: failureColors,
+  [TaskStatus.TestTimedOut]: failureColors,
+  [TaskStatus.SetupFailed]: {
+    border: "#E7DBEC",
+    fill: "#F3EDF5",
+    text: "#877290"
+  }
+};
+
+interface BadgeColorProps {
+  border: string;
+  fill: string;
+  text: string;
+}
+
+// only use for statuses whose color is not supported by leafygreen badge variants, i.e. SystemFailed, TestTimedOut, SetupFailed
+const StyledBadge = styled(Badge)`
+  border-color: ${(props: BadgeColorProps) => props.border} !important;
+  background-color: ${(props: BadgeColorProps) => props.fill} !important;
+  color: ${(props: BadgeColorProps) => props.text} !important;
+`;
+
+const renderStatusBadge = (status: string) => {
+  if (status in mapTaskStatusToBadgeVariant) {
+    return (
+      <Badge key={status} variant={mapTaskStatusToBadgeVariant[status]}>
+        {status}
+      </Badge>
+    );
+  } else if (status in mapUnsupportedBadgeColors) {
+    return (
+      <StyledBadge key={status} {...mapUnsupportedBadgeColors[status]}>
+        {status}
+      </StyledBadge>
+    );
+  }
+  throw new Error(`Status '${status}' is not a valid task status`);
```

Comment:
I think we should have an error boundary around the task table so the user can continue using the App if this error is thrown.

Reply:
an error boundary would just contain the error to the task table, it would not make it silent. i decided to throw an error here because really this should never happen and if it does it should not be silent - on the contrary it should be loud. i made a ticket to report errors and their severity to bugsnag, which would be what is ideally done here: https://jira.mongodb.org/browse/EVG-7645

Reply:
Since we send errors to bugsnag, I don't think we need to sacrifice UX for error reporting. In the case that this error is thrown, the user won't be able to use the application because they will see an error screen. We have the Bugsnag error boundary component already installed in our App for this purpose.